### ISSUE TITLE
model binding multiple sources to a single class doesn't work as doumented

### ISSUE THREAD

**Issue by KillerBoogie** [dotnet/AspNetCore.Docs#29295]
The first example under the section sources shows that you can mix various sources into one model. This does not work! 

E.g. 
```
[HttpPost]
public ActionResult Post(TestDetails testDetails)
{
    return Ok();
}

public record TestDetails
{
    [FromBody]
    public Artist? Artist { get; init; }

    [ModelBinder(typeof(EnvironmentBinder))]
    public IPAddress? IpAddress { get; init; }

    [FromHeader(Name = "Accept-Language")]
    public string? PreferredLanguages { get; init; }

    [FromQuery]
    public string? SelectedLanguage { get; init; }
}

public record Body
{
    public string? Name { get; init; }

    public string? StageName { get; init; }
}
```
Calling the endpoint creates an error stating that Artist must not be empty.

To make it work either the following option must be set

```
builder.Services.AddMvc().ConfigureApiBehaviorOptions(options => {
    options.SuppressInferBindingSourcesForParameters = true;
});

```
or (from .Net 6 on?) the method parameter must be declared as [FromQuery]:

```
[HttpPost]
public ActionResult Post([FromQuery] TestDetails testDetails)
{
    return Ok();
}
```

This is totally missing in the documentation and cost me several days to figure out.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: e24f48f8-6c4d-ecda-6ce6-a7826d396e99
* Version Independent ID: 9c4af557-8d06-4c6c-ad2a-0dd234547563
* Content: [Model Binding in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-7.0)
* Content Source: [aspnetcore/mvc/models/model-binding.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/models/model-binding.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
None of the code you posted is in this tutorial.

Please reproduce a problem with the code in this tutorial.

**Comment by KillerBoogie** [comment]
The code I posted is an extended example to show the problem.

In the article at section [Sources](https://learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-7.0#sources) the example shows:

```
public class Instructor
{
    public int Id { get; set; }

    [FromQuery(Name = "Note")]
    public string? NoteFromQueryString { get; set; }

    // ...
}
```
When using the class in a controller method like this...

```
[HttpPost]
public ActionResult PostOneClass(Instructor instructor)
{
    return Ok(instructor);
}
```
...the property `NoteFromQueryString`  is not set. It is always null. So the example does not work!

When the instructor class is attributed with a `[FromQuery]` ...
 ```
[HttpPost]
public ActionResult PostOneClass([FromQuery] Instructor instructor)
{
    return Ok(instructor);
}
```
...the `NoteFromQueryString` property is set, but not the `Id`.

When the `Id` is marked additionally as `[FromBody]` ...
```
public class Instructor
{
    [FromBody]
    public int Id { get; set; }

    [FromQuery(Name = "Note")]
    public string? NoteFromQueryString { get; set; }

}
```
...a 400 error is returned, stating: "The JSON value could not be converted to System.Int32..."

Keeping `[FromBody]` in the Instructor class and removing `[FromQuery]` from the controller method makes `NoteFromQueryString` return null again.

**The only solution I found is to put the body parameter(s) in a separate class...**

```
public class Instructor
{
    [FromBody]
    public Body body { get; set; }

    [FromQuery(Name = "Note")]
    public string? NoteFromQueryString { get; set; }

}
public class Body
{
    public int Id { get; set; }
   //...
}
```

**... and use `[FromQuery]` in the method:**

```
[HttpPost]
public ActionResult PostOneClass([FromQuery] Instructor instructor)
{
    return Ok(instructor);
}
```

Without `[FromQuery]` in the method a request returns a 400 error, stating: "The body field is required."

So, the example doesn't work and the only working solution is not presented.

**Comment by Rick-Anderson** [comment]
@serpent5 @sammychinedu2ky @fiyazbinhasan  please review

**Comment by sammychinedu2ky** [comment]
@Rick-Anderson  @KillerBoogie 

 https://learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-7.0#frombody-attribute
 
![image](https://github.com/dotnet/AspNetCore.Docs/assets/36219292/4b1ba5d7-1c38-4015-84bd-045adbc8d499)

**Comment by KillerBoogie** [comment]
Yes, this is correctly stated. But, even if '[FromBody]' is not stated as in the example above it doesn't work either, because it is added as a convention. I just listed all options and outcomes in my post.

My main point is that there is no way of getting the stated example working with the information given in the article. The critical information to make this example work is not obvious and no where found in the article. It is only stated that it won't work with `[FromBody]', but not how to mae it work. How do you make this example from the sources section above work? Please tell me and then state it in the article.

```
public class Instructor
{
    public int Id { get; set; }

    [FromQuery(Name = "Note")]
    public string? NoteFromQueryString { get; set; }

    // ...
}
```
I even didn't know that it is possible before I saw the example and then failed when trying it out. I lost a week of my time just to figure out how it works by testing all options instead of reading a good documentation for 10 mins.

Here is another option, creating a custom `[RequestAttribute]`, as a workaround: https://josef.codes/model-bind-multiple-sources-to-a-single-class-in-asp-net-core/

**Comment by sammychinedu2ky** [comment]
@KillerBoogie
